presto-main/src/main/java/com/facebook/presto/cost/CostCalculatorUsingExchanges.java

```diff
@@ -119,6 +121,12 @@ public PlanNodeCostEstimate visitGroupReference(GroupReference node, Void contex
             throw new UnsupportedOperationException();
         }
 
+        @Override
+        public PlanNodeCostEstimate visitAssignUniqueId(AssignUniqueId node, Void context)
+        {
+            return cpuCost(getStats(node).getOutputRowCount() * BigintType.BIGINT.getFixedSize());
```

Comment:
Why are you hardcoding `BigintType.BIGINT.getFixedSize()` instead of `getStats(node).getSymbolStatistics(node.getIdColumn()).getAverageRowSize()`?

Reply:
AFAIK we don't fill `#getAverageRowSize()` based on type yet

Reply:
@findepi It looks like we do. `AssignUniqueIdStatsRule` contains `.setAverageRowSize(BIGINT.getFixedSize())`.

Reply:
Right. I believe my comment remains generally applicable, although in this case we could defer to average row size from stats.